src/api/v1/friend.ts

```diff
@@ -70,6 +70,28 @@ export const updateFriend = async (req: Request, res: Response) => {
 		res.status(404).send();
 		return;
 	}
+	if (setBody["getFrontNotif"] === false) {
+		await getCollection("friends").updateOne(
+			{
+				uid: req.params.id,
+				frienduid: res.locals.uid,
+				$or: [{ lastOperationTime: null }, { lastOperationTime: { $lte: res.locals.operationTime } }],
+			},
+			{ $set: { "getTheirFrontNotif": false } }
+		);
+	}
+	if (setBody["getTheirFrontNotif"] === true) {
```

Comment:
Wouldn't this already be set in the database on line 61? Should we not check if we can set this to true before we set it on line 61?

Also to prevent the app from failing to set any data (receiving 403) we should instead force the value to be false rather than fail. The app would still have the value as true for all already-existing data, and would therefore send that property as true if it didn't change (I believe, would need to check). And I don't want the app to break if we eventually merge this.

Reply:
Totally correct, new commit soon. We're going to keep the first check (`getFrontNotif === false`) following the db update to avoid trying to make more changes if the original operation fails (404)